setup.py

```diff
@@ -250,12 +263,74 @@ def run(self):
         self.distribution.data_files = get_data_files()
 
 
+class JSX(BaseCommand):
+    description = "build admin app"
+
+    jsx_dir = pjoin(here, 'jsx')
+    js_target = pjoin(static, 'js', 'admin-react.js')
+
+    def should_run(self):
+        if os.getenv('READTHEDOCS'):
+            # yarn not available on RTD
+            return False
+
+        if not os.path.exists(self.js_target):
+            return True
+
+        js_target_mtime = mtime(self.js_target)
+        # The sdist doesn't contain the jsx directory so this always returns 0
```

Comment:
I think this means we need have a test for this in release.yml. The sdist definitely should contain the `jsx` sources. I think this means MANIFEST.in needs

```
graft jsx
prune jsx/build
prune jsx/node_modules
```

Reply:
Added, including a test in release.yml.